shell/src/main/java/alluxio/cli/MetricsDocGenerator.java

```diff
@@ -0,0 +1,162 @@
+  private static final String[] CATEGORIES = new String[]{"cluster", "master", "worker", "client"};
+  private static final String CSV_FILE_DIR = "docs/_data/table/";
+  private static final String YML_FILE_DIR = "docs/_data/table/en/";
+  private static final String CSV_SUFFIX = "csv";
+  private static final String YML_SUFFIX = "yml";
+  private static final String CSV_FILE_HEADER = "metricName,metricType";
+
+  /**
+   * Writes the supported files for metrics system doc.
+   */
+  private static void writeDocFiles() throws IOException {
+    // Gets and sorts the metric keys
+    List<MetricKey> defaultKeys = new ArrayList<>(MetricKey.allMetricKeys());
+    Collections.sort(defaultKeys);
+
+    String homeDir = new InstancedConfiguration(ConfigurationUtils.defaults())
+        .get(PropertyKey.HOME);
+
+    // Map from metric key prefix to metric category
+    Map<String, String> metricTypeMap = new HashMap<>();
+    for (MetricsSystem.InstanceType type : MetricsSystem.InstanceType.values()) {
+      String typeStr = type.toString();
+      String category = typeStr.toLowerCase();
+      metricTypeMap.put(typeStr, category);
+    }
+
+    try (Closer closer = Closer.create()) {
+      Map<FileWriterKey, FileWriter> fileWriterMap = new HashMap<>();
+      String csvFolder = PathUtils.concatPath(homeDir, CSV_FILE_DIR);
+      String ymlFolder = PathUtils.concatPath(homeDir, YML_FILE_DIR);
+      FileWriter csvFileWriter;
+      FileWriter ymlFileWriter;
+      for (String category : CATEGORIES) {
+        csvFileWriter = new FileWriter(PathUtils
+            .concatPath(csvFolder, category + "-metrics." + CSV_SUFFIX));
+        csvFileWriter.append(CSV_FILE_HEADER + "\n");
+        ymlFileWriter = new FileWriter(PathUtils
+            .concatPath(ymlFolder, category + "-metrics." + YML_SUFFIX));
+        fileWriterMap.put(new FileWriterKey(category, CSV_SUFFIX), csvFileWriter);
+        fileWriterMap.put(new FileWriterKey(category, YML_SUFFIX), ymlFileWriter);
+        //register file writer
+        closer.register(csvFileWriter);
+        closer.register(ymlFileWriter);
+      }
+
+      for (MetricKey metricKey : defaultKeys) {
+        String key = metricKey.toString();
+
+        String[] components = key.split("\\.");
+        if (components.length < 2) {
+          throw new IOException(String
+              .format("The given metric key %s doesn't have two or more components", key));
+        }
+        if (metricTypeMap.containsKey(components[0])) {
+          csvFileWriter = fileWriterMap.get(
+              new FileWriterKey(metricTypeMap.get(components[0]), CSV_SUFFIX));
+          ymlFileWriter = fileWriterMap.get(
+              new FileWriterKey(metricTypeMap.get(components[0]), YML_SUFFIX));
+          csvFileWriter.append(String.format("%s,%s%n", key, metricKey.getMetricType().toString()));
+          ymlFileWriter.append(String.format("%s,%s%n",
+              key, metricKey.getDescription().replace("'", "''")));
```

Comment:
Can we also use the `escapeHtml` function (see confDocGen for usage) so that it doesn't break any doc pages if someone adds some weird characters to a metric description

Reply:
Add the `escapeHtml`, thanks!